src/CHIPDriver.hh

```diff
@@ -118,4 +120,183 @@ extern std::atomic_ulong CHIPNumRegisteredFatBinaries;
+    case OpenCL:
+      return "opencl";
+    case Level0:
+      return "level0";
+    case Default:
+      return "default";
+    default:
+      assert(!"Unknown backend type!");
+      return "unknown";
+    }
+  }
+
+  Type getType() const { return Type_; }
+};
+
+class EnvVars {
+private:
+  int PlatformIdx_ = 0;
+  DeviceType Device_ = DeviceType::GPU;
+  int DeviceIdx_ = 0;
+  BackendType Backend_ = BackendType::Default;
+  bool DumpSpirv_ = false;
+  std::string JitFlags_ = CHIP_DEFAULT_JIT_FLAGS;
+  bool L0ImmCmdLists_ = true;
+  int L0CollectEventsTimeout_ = 0;
+
+public:
+  EnvVars() : Device_(DeviceType::GPU), Backend_(BackendType::Default) {
+    parseEnvironmentVariables();
+    logDebugSettings();
+  }
+
+  int getPlatformIdx() const { return PlatformIdx_; }
+  DeviceType getDevice() const { return Device_; }
+  int getDeviceIdx() const { return DeviceIdx_; }
+  BackendType getBackend() const { return Backend_; }
+  bool getDumpSpirv() const { return DumpSpirv_; }
+  const std::string &getJitFlags() const { return JitFlags_; }
+  bool getL0ImmCmdLists() const { return L0ImmCmdLists_; }
+  int getL0CollectEventsTimeout() const { return L0CollectEventsTimeout_; }
+
+private:
+  void parseEnvironmentVariables() {
+    // Parse all the environment variables and set the class members
+    if (!readEnvVar("CHIP_PLATFORM").empty()) {
+      PlatformIdx_ = parseInt("CHIP_PLATFORM");
+      Device_ = parseDeviceType("CHIP_DEVICE_TYPE");
+    }
+    if (!readEnvVar("CHIP_DEVICE").empty()) {
+      DeviceIdx_ = parseInt("CHIP_DEVICE");
+      Backend_ = parseBackendType("CHIP_BE");
+    }
+    if (!readEnvVar("CHIP_DUMP_SPIRV").empty()) {
+      DumpSpirv_ = parseBoolean("CHIP_DUMP_SPIRV");
+      JitFlags_ = parseJitFlags("CHIP_JIT_FLAGS_OVERRIDE");
+    }
+    if (!readEnvVar("CHIP_L0_IMM_CMD_LISTS").empty())
+      L0ImmCmdLists_ = parseBoolean("CHIP_L0_IMM_CMD_LISTS");
+    if (!readEnvVar("CHIP_L0_COLLECT_EVENTS_TIMEOUT").empty())
+      L0CollectEventsTimeout_ = parseInt("CHIP_L0_COLLECT_EVENTS_TIMEOUT");
+  }
+
+  std::string_view parseJitFlags(const std::string &StrIn) {
+    if (readEnvVar(StrIn).empty())
+      return CHIP_DEFAULT_JIT_FLAGS;
+
+    return JitFlags_;
+  }
+
+  int parseInt(const std::string &StrIn) {
+    const auto &Str = readEnvVar(StrIn);
+    if (!isConvertibleToInt(Str))
+      CHIPERR_LOG_AND_THROW("Invalid integer value: " + Str,
+                            hipErrorInitializationError);
+    return std::stoi(str);
+  }
+
+  bool parseBoolean(const std::string &StrIn) {
+    auto str = readEnvVar(StrIn);
+    if (str == "1" || str == "on")
+      return true;
+    if (str == "0" || str == "off")
+      return false;
+    CHIPERR_LOG_AND_THROW("Invalid boolean value: " + str + "while parsing " +
+                              StrIn,
+                          hipErrorInitializationError);
+    return false; // This return is never reached
+  }
+
+  DeviceType parseDeviceType(const std::string &StrIn) {
```

Comment:
Looks like there is a separation of concerns issue. The concern of constructing DeviceType from a string looks something the class should do. E.g. via `static DeviceType fromString(const std::string&);`.